qcengine/programs/entos.py

```
@@ -0,0 +1,238 @@
+                exc["proc"].wait(timeout=5)
+            self.version_cache[which_prog] = safe_version(exc["stdout"].split()[2])
+
+        return self.version_cache[which_prog]
+
+    def compute(self, input_data: 'ResultInput', config: 'JobConfig') -> 'Result':
+        """
+        Run entos
+        """
+        # Check if entos executable is found
+        self.found(raise_error=True)
+
+        # Check entos version
+        if parse_version(self.get_version()) < parse_version("0.5"):
+            raise TypeError("entos version '{}' not supported".format(self.get_version()))
+
+        # Setup the job
+        job_inputs = self.build_input(input_data, config)
+
+        # Run entos
+        proc = self.execute(job_inputs)
+
+        if isinstance(proc, FailedOperation):
+            return proc
+        else:
+            # If execution succeeded, collect results
+            result = self.parse_output(proc["outfiles"], input_data)
+            return result
+
+    def execute(self, inputs, extra_outfiles=None, extra_commands=None, scratch_name=None, timeout=None):
+
+        if extra_outfiles is not None:
+            outfiles = ["dispatch.out"].extend(extra_outfiles)
+        else:
+            outfiles = ["dispatch.out"]
+
+        if extra_commands is not None:
+            commands = extra_commands
+        else:
+            commands = inputs["commands"]
+
+        exe_success, proc = execute(commands,
+                                    infiles=inputs["infiles"],
+                                    outfiles=outfiles,
+                                    scratch_location=inputs["scratch_location"],
+                                    scratch_name=scratch_name,
+                                    timeout=timeout
+                                    )
+        proc["outfiles"]["dispatch.out"] = proc["stdout"]
+
+        # Determine whether the calculation succeeded
+        output_data = {}
+        if not exe_success:
```

You can use the new `raise UnknownError` commands rather than forming a full `FailedOperation` see Psi4/rdkit.